vessim/data.py

```diff
@@ -0,0 +1,169 @@
+        use_forecast: Bool indicating if forecast should be loaded. Default is true.
+
+    Returns:
+        The dataframe of actual data, the optional dataframe of forecast data and the
+        fill_method to be fed into a TimeSeriesApi.
+
+    Raises:
+        RuntimeError if dataset can not be loaded.
+    """
+    if isinstance(dataset, str):
+        dataset_dict: Dict = Datasets[dataset]
+    else:
+        dataset_dict = dataset
+
+    required_files = [dataset_dict["actual"]]
+    if use_forecast:
+        required_files.append(dataset_dict["forecast"])
+
+    dir_path = Path(data_dir or "").expanduser().resolve()
+
+    if not _check_files(required_files, dir_path):
+        if "url" not in dataset_dict.keys():
+            raise RuntimeError("Data files could not be found.")
+
+        print("Required data files not present. Try downloading...")
+        os.makedirs(dir_path, exist_ok=True)
+
+        try:
+            urllib.request.urlretrieve(dataset_dict["url"], dir_path / "dataset.zip")
+        except Exception:
+            raise RuntimeError(
+                f"Dataset could not be retrieved from url: {dataset_dict['url']}"
+            )
+
+        with ZipFile(dir_path / "dataset.zip", "r") as zip_ref:
+            zip_ref.extractall(path=dir_path)
+        os.remove(dir_path / "dataset.zip")
+        print("Successfully downloaded and unpacked data files.")
+
+    actual = _read_data_from_csv(
+        dir_path / dataset_dict["actual"], index_cols=[0], scale=scale
+    )
+
+    forecast: Optional[PandasObject] = None
+    if use_forecast:
+        if dataset_dict.get("static_forecast", False):
+            # There is only one timestamp present in the forecast (static forecast)
+            index_cols: List[int] = [0]
+        else:
+            # There are two timestamps present in the forecast (non-static forecast)
+            index_cols = [0, 1]
+
+        forecast = _read_data_from_csv(
+            dir_path / dataset_dict["forecast"], index_cols=index_cols, scale=scale
+        )
+
+    if start_time is not None:
+        shift = pd.to_datetime(start_time) - actual.index[0]
+        actual.index += shift
+        if use_forecast:
+            forecast = _shift_dataframe(forecast, shift) # type: ignore
+
+    return actual, forecast, dataset_dict.get("fill_method", "bfill")
+
+
+def _check_files(files: List[str], base_dir: Path) -> bool:
+    """Check whether files are present in specified base directory."""
+    for file in files:
+        path = os.path.join(base_dir, file)
+        if not os.path.isfile(path):
+            return False
+    return True
+
+
+def _read_data_from_csv(
+    path: Path, index_cols: List[int], scale: float = 1.0
+) -> PandasObject:
+    """Retrieves a dataframe from a csv file and transforms it."""
+    df = convert_to_datetime(pd.read_csv(path, index_col=index_cols))
+    return (df * scale).astype(float)
+
+
+def convert_to_datetime(df: PandasObject) -> PandasObject:
```

Comment:
shouldn't this be private? And maybe lets rename to `_convert_index_to_datetime`